ci/generate_changelog.py

```diff
@@ -0,0 +1,203 @@
+    )
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Show the changes without prepending them to the changelog file.",
+    )
+
+    return parser.parse_args()
+
+
+def get_old_series_name(out_file: str) -> str:
+    """Function that finds and returns the old series name of the previous version
+    so that it can be used as the default value.
+
+    Args:
+        out_file (str): Path to the changelog.md file.
+
+    Returns:
+        str: The old series name.
+    """
+
+    series_name = ""
+
+    with open(out_file, "r", encoding="utf-8") as changelog_file:
+        old_entries = changelog_file.read()
+        match = regex.search(PREVIOUS_VERSION_REGEX, old_entries)
+        if match:
+            line_after_version = old_entries[match.end() :].splitlines()[0]
+            match = regex.search(SERIES_NAME_REGEX, line_after_version)
+            if match:
+                series_name = match.group() + "\n"
+
+    return series_name
+
+
+def gather_output(
+    unreleased_dir: str, out_version: str, out_series_name: str, dry_run: bool
+) -> Optional[str]:
+    """Function that handles gathering all the changelog entries from the unreleased dir
+    into a string. Returns None, if there are no entries.
+
+    Args:
+        unreleased_dir (str):   path into (and including) .unreleased directory where all the
+                                files with the change descriptions are stored.
+        out_version (str):      Version of the release that's being made.
+        out_series_name (str):  Version series name (like android lollipop).
+        dry_run (bool):         Only show the changes that would be made and don't write to
+                                or delete any files.
+
+    Returns:
+        Optional[str]: The aggregated entries including the header and footer to
+                       prepend to the changelog file.
+    """
+
+    # The following bool is used to prevent unwanted behavior
+    # when no version entries are present in the new version directory.
+    at_least_one_entry_found = False
+
+    output = HEADER_FORMAT.format(out_version, out_series_name)
+
+    for changelog_entry_file in os.scandir(unreleased_dir):
+        with open(changelog_entry_file, "r", encoding="utf-8") as changelog_entry_file:
+            for line in changelog_entry_file.readlines():
+                at_least_one_entry_found = True
+                entry_ticket_id = os.path.basename(changelog_entry_file.name)
+                output += ENTRY_FORMAT.format(entry_ticket_id, line.strip())
+        if not dry_run:
+            os.remove(changelog_entry_file.name)
+    output += VERSION_ENDING
+
+    return output if at_least_one_entry_found else None
+
+
+def generate_changelog(
+    unreleased_dir: str,
+    out_version: str,
+    out_series_name: Optional[str],
+    out_file: str,
+    dry_run: bool,
+) -> int:
+    """Main function for handling the changelog generation."""
```

Comment:
To make it a bit more robust, maybe it would make sence setting pocess workdir to dir(__file__).pop() ?
So if called not from root would still work as intended?

Reply:
Done